### PR TITLE
Made output prettier and more understandable

### PR DESCRIPTION
**Pull request by mouse07410**
Test output of this modified botan.py is easier to follow.

### CODE REVIEW

File: src/python/botan.py [hunk 1]
```diff
@@ -15,7 +16,7 @@
 """
 Module initialization
 """
-botan = CDLL('libbotan-1.11.so')
+botan = CDLL('libbotan-1.11.dylib')
```
Comment (webmaster128) [inline src/python/botan.py]: I think this should be somehow platform independent. Does that work in Windows as well?

Comment (mouse07410) [inline src/python/botan.py]: Oops! My fault - it is not platform-independent. Thanks for catching it. Will fix and re-post.

Regarding "random links" - I thought that a Pull Request with GitHub comments is clear enough? Or should I duplicate on the mailing list what I already put in the GitHub comments? No big deal for me (cut-n-paste works), just want to know what your preference is.

Sent from my iPad

On Aug 12, 2015, at 02:38, Simon Warta <notifications@github.com<mailto:notifications@github.com>> wrote:


In src/python/botan.py<https://github.com/randombit/botan/pull/246#discussion_r36830343>:

> @@ -15,7 +16,7 @@
>  """
>  Module initialization
>  """
> -botan = CDLL('libbotan-1.11.so')
> +botan = CDLL('libbotan-1.11.dylib')
>


I think this should be somehow platform independent. Does that work in Windows as well?

Comment (mouse07410) [inline src/python/botan.py]: You are correct. I need to restore this to 

botan = CDLL('libbotan-1.11.so')

Comment (mouse07410) [inline src/python/botan.py]: Here’s an update PR: https://github.com/randombit/botan/pull/247
It fixes the issue introduced by the previous one (#246).

Here’s a copy of the PR comment:


Fixed introduced Mac-specific change, returning the patch to platform-independent.

Original patch was to improve clarity and readability of the test/demo output provided by botan.py.

I don’t think more details are needed for such a small (and fairly obvious) patch?


On Aug 12, 2015, at 2:38 , Simon Warta <notifications@github.com<mailto:notifications@github.com>> wrote:


In src/python/botan.py<https://github.com/randombit/botan/pull/246#discussion_r36830343>:

> @@ -15,7 +16,7 @@
>  """
>  Module initialization
>  """
> -botan = CDLL('libbotan-1.11.so')
> +botan = CDLL('libbotan-1.11.dylib')
>


I think this should be somehow platform independent. Does that work in Windows as well?

—
Reply to this email directly or view it on GitHub<https://github.com/randombit/botan/pull/246/files#r36830343>.

--
Uri Blumenthal
uri@mit.edu<mailto:uri@mit.edu>

**Comment by webmaster128** [comment]
Thanks for targeting this. I'll check the follow-up PR soon.

> Regarding "random links" - I thought that a Pull Request with GitHub comments is clear enough? Or should I duplicate on the mailing list what I already put in the GitHub comments?

Yes we all see your pull request and will review it earlier or later.

My point was that the place where you posted the link was entirely unrelated to the content of the PR. It was a discussion about a change in a C-API test. There are issue, pull requests and comments on commits on Github. Please check where you are before responding. I think the best would be to avoid using an email client to respond on Github because you fill entire screens with your email footer and quotes. Isn't there a Github client for the iPad?

**Comment by mouse07410** [comment]
1. Point taken.
2. Apparently there are GitHub clients for iOS. Investigating one now

Sent from my iPad

On Aug 12, 2015, at 10:06, Simon Warta <notifications@github.com<mailto:notifications@github.com>> wrote:


Thanks for targeting this. I'll check the follow-up PR soon.

Regarding "random links" - I thought that a Pull Request with GitHub comments is clear enough? Or should I duplicate on the mailing list what I already put in the GitHub comments?

Yes we all see your pull request and will review it earlier or later.

My point was that the place where you posted the link was entirely unrelated to the content of the PR. It was a discussion about a change in a C-API test. There are issue, pull requests and comments on commits on Github. Please check where you are before responding. I think the best would be to avoid using an email client to respond on Github because you fill entire screens with your email footer and quotes. Isn't there a Github client for the iPad?

**Comment by mouse07410** [comment]
One more question/clarification: since GitHub notifies maintainers when a PR against their repo is created - does it make sense to post a URL to that PR here? Or better just keep it all within the GitHub?

Sent from my iPad

On Aug 12, 2015, at 10:06, Simon Warta <notifications@github.com<mailto:notifications@github.com>> wrote:


Thanks for targeting this. I'll check the follow-up PR soon.

Regarding "random links" - I thought that a Pull Request with GitHub comments is clear enough? Or should I duplicate on the mailing list what I already put in the GitHub comments?

Yes we all see your pull request and will review it earlier or later.

My point was that the place where you posted the link was entirely unrelated to the content of the PR. It was a discussion about a change in a C-API test. There are issue, pull requests and comments on commits on Github. Please check where you are before responding. I think the best would be to avoid using an email client to respond on Github because you fill entire screens with your email footer and quotes. Isn't there a Github client for the iPad?

**Comment by webmaster128** [comment]
From your understanding of *hear* I not get the idea that you are entirely mixing up Github-conversations and mailing list emails. That would explain a lot ;) Right now, *here* is the comments to your first, closed pull request: https://github.com/randombit/botan/pull/246

Please note that notifications to Github content has [botan] in the subject line and comes from notifications@github.com whereas mailing list emails contain [botan-devel] in the subject and come from it's original sender.

To answer your question: maintainers get notified about every single word that someone writes related to the project. So no need to inform someone about pull requests. Additionally there is a nice overview of all pull requests which I check regularly during my own work: https://github.com/randombit/botan/pulls

**Comment by mouse07410** [comment]
Thank you - you’re correct: I did not entirely understand the conversations mechanism. :-)   I think now I understand how “it” works much better (though probably still less than perfectly :).

On Aug 12, 2015, at 17:25 , Simon Warta <notifications@github.com<mailto:notifications@github.com>> wrote:

From your understanding of hear I not get the idea that you are entirely mixing up Github-conversations and mailing list emails. That would explain a lot ;) Right now, here is the comments to your first, closed pull request: #246<https://github.com/randombit/botan/pull/246>

Please note that notifications to Github content has [botan] in the subject line and comes from notifications@github.com<mailto:notifications@github.com> whereas mailing list emails contain [botan-devel] in the subject and come from it's original sender.

To answer your question: maintainers get notified about every single word that someone writes related to the project. So no need to inform someone about pull requests. Additionally there is a nice overview of all pull requests which I check regularly during my own work: https://github.com/randombit/botan/pulls

—
Reply to this email directly or view it on GitHub<https://github.com/randombit/botan/pull/246#issuecomment-130453330>.

--
Uri Blumenthal
uri@mit.edu<mailto:uri@mit.edu>

**Pull request closed without merging**